design-proposals/isolate-emulator-thread-with-smt-fix/isolate-emulator-thread-with-smt-fix.md

```diff
@@ -54,13 +54,63 @@ Fix the actual scheduling of the virt-launcher to a node that will reject it.
 This bug addresses all KubeVirt users running VMs with isolateEmulatorThread on nodes with hyperthreading enabled - specifically DPDK and Realtime users.
 
 ## User Stories
-A user should be able to run VM with dedicatedCPUPlancement and isolateEmulatorThread enabled on a node with hyperthreading enabled, regardless of the amount of guest CPUs requested.
+* As a cluster admin, I want to set the cluster-wide IsolatedEmulatorThread behavior for Kubevirt by setting a knob on HCO.
+* As a Kubevirt user, I should be able to run VM with dedicatedCPUPlancement and isolateEmulatorThread enabled on a node with hyperthreading enabled, regardless of the amount of guest CPUs requested.
 
 ## Repos
 - [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+- [kubevirt/hyperconverged-cluster-operator](https://github.com/kubevirt/hyperconverged-cluster-operator)
 
 # Bug fix design proposal
-The bug fix proposal introduces an alpha annotation `kubevirt.io/CPUManagerPolicyBetaOptions:full-pcpus-only`:
+The bug fix proposal introduces an alpha annotation to the kubevirt CR: `kubevirt.io/EmulatorThreadEvenParity`:
+```
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  annotations:
+    kubevirt.io/latest-observed-api-version: v1
+    kubevirt.io/storage-observed-api-version: v1
+    kubevirt.io/EmulatorThreadEvenParity:
+spec:
+  configuration:
+    developerConfiguration:
+      featureGates:
+      - EmulatorThreadEvenParity
+```
+This annotation will be supported only when an appropriate featureGate is enabled: EmulatorThreadEvenParity.
+
+When the following is configured:
+1. kubevirt CR has the annotation.
+2. kubevirt CR enabled the feature gate.
+3. VM is configured with `isolateEmulatorThread: true`
+
+then the virt-controller will:
+1. Set the virt-launcher pod with an even amount of CPUs, meaning:
+
+| User requested CPUs               |Even (2,4,6,8,...)| odd (1,3,4,5,7,9,...) | Notes                                                                       |
+|-----------------------------------|------------------|-----------------------|-----------------------------------------------------------------------------|
+| Extra emulatorThread CPUs for pod |        +2        | +1 (as usual)         | * In both scenarios the guest will get the same amount of CPUs it requested | 
+
+2. Copy the annotation to the VMI spec in order for it to be used on the virt-launcher, when setting the housekeeping cgroup. Doing so avoids adding API on the virt-handler to virt-launcher grpc protocol.
```

Comment:
> In our discussions we have talked about making the VMI mutating webhook adding the annotation to the VMI (in order for the VMI-controller to just react to its presence). The VMI mutating webhook should add the annotation in case the following conditions are met:
> 
> 1. The FG is enabled.
> 2. The annotation is present on the KubeVirt CR.
> 3. The VMI is defined with IsolateEmulatorThread option enabled.

You're right, will fix. 
I've also set the conditions that needs to be met on every step.